stan/math/rev/functor/coupled_ode_system.hpp

```diff
@@ -400,27 +83,48 @@ struct coupled_ode_system<F, var, var> {
    *
    * @param[in] f the base ODE system functor
    * @param[in] y0 the initial state of the base ode
-   * @param[in] theta parameters of the base ode
-   * @param[in] x real data
-   * @param[in] x_int integer data
    * @param[in, out] msgs stream for messages
+   * @param[in] args other additional arguments
    */
-  coupled_ode_system(const F& f, const std::vector<var>& y0,
-                     const std::vector<var>& theta,
-                     const std::vector<double>& x,
-                     const std::vector<int>& x_int, std::ostream* msgs)
+  coupled_ode_system_impl(const F& f,
```

Comment:
It's much better to have a single coupled_ode_system_impl thing rather than all those specialisations! I wanted this simplification for quite some time.